bulk_processor.go

```diff
@@ -545,6 +545,7 @@ func (w *bulkWorker) waitForActiveConnection() {
 	deadline := time.NewTimer(w.p.stopRecon)
 	interrupt := make(chan os.Signal, 1)
```

Comment:
What are you trying to achieve with signals in Elastic?

Reply:
In my test case I have a main process that is listening on an event source and using these events to add work to an Elastic bulk processor.  This main process also listens for Ctrl-C signals in another go routine and in such a case exits the event loop and calls Close on the bulk processor.  
Now, the reason I added signals to this PR is that when Elastic enters this new waiting state, calls to both Add and Close on the processor will block.  Thus in my test case, with Elasticsearch down, I could no longer quit the main process with Ctrl-C since this was calling Close() but that blocks forever.  So I needed a way for waitForConnections to exit without needing to restart the Elasticsearch server.  
Maybe I could have achieved this by adding a another channel to the bulk processor and pushing a message to it when Close() was called.

Reply:
I see. Then we need to change `Close` in a way that it doesn't block, or at least times out without relying on signals.

I see your use case. But I think there should be a more general mechanism (besides Ctrl-C) to stop/close bulk processor cleanly. I see signals typically used with console-based apps, and don't see a good reason to rely on them outside of `main`.